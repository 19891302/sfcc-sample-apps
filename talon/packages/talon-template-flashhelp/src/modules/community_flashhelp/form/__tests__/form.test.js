import { createElement } from 'lwc';
import Form from 'community_flashhelp/form';

jest.mock("talon/routingService", () => {
    return {
        navigateToRoute: jest.fn()
    };
});

global.fetch = jest.fn();

describe('community_flashhelp/form', () => {
    beforeEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }

        jest.resetAllMocks();

        fetch.mockReturnValue(Promise.resolve());
    });

    it('renders correctly', () => {
        const element = createElement('community_flashhelp-form', { is: Form });
        document.body.appendChild(element);

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
