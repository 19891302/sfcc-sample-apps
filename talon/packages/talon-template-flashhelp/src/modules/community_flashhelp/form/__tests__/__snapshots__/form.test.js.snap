// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`community_flashhelp/form renders correctly 1`] = `
<community_flashhelp-form>
  #shadow-root(open)
    <div
      class="wrap"
    >
      <div
        class="main"
      >
        <div
          class="slds-grid slds-m-vertical_x-large"
        >
          <div
            class="slds-col slds-size_4-of-12"
          />
          <div
            class="slds-col slds-size_4-of-12"
          >
            <div
              class="slds-grid form-title"
            >
              <div
                class="slds-col slds-size_1-of-12"
              >
                <span
                  class="slds-icon_container slds-icon-standard-account"
                  title="Hurricane Harvey Immediate Assistance Program"
                >
                  <svg
                    aria-hidden="true"
                    class="slds-icon"
                  >
                    <use
                      xlink:href="assets/icons/standard-sprite/svg/symbols.svg#custom_notification"
                    />
                  </svg>
                  <span
                    class="slds-assistive-text"
                  >
                    Hurricane Harvey Immediate Assistance Program
                  </span>
                </span>
              </div>
              <div
                class="slds-col slds-size_11-of-12"
              >
                <h4>
                  Hurricane Harvey Immediate Assistance Program
                </h4>
              </div>
            </div>
            <div
              class="slds-form slds-form_stacked"
            >
              <lightning-input
                class="immediate-assistance-name"
              >
                #shadow-root(open)
                  <input
                    aria-label="Name"
                    data-readonly="false"
                    data-required="true"
                    placeholder="Your name here"
                  />
              </lightning-input>
              <lightning-input-address
                class="immediate-assistance-address"
              >
                #shadow-root(open)
                  <input
                    data-address-label="Address"
                    data-city-label="City"
                    data-country-label="Country"
                    data-postal-code-label="Zip"
                    data-province-label="State / Province"
                    data-street-label="Street"
                    required=""
                  />
              </lightning-input-address>
              <lightning-input
                class="immediate-assistance-telephone"
              >
                #shadow-root(open)
                  <input
                    aria-label="Telephone"
                    data-readonly="false"
                    data-required="true"
                    placeholder="eg: (555) 867-5309"
                  />
              </lightning-input>
              <lightning-input
                class="immediate-assistance-email"
              >
                #shadow-root(open)
                  <input
                    aria-label="Global_Address.email"
                    data-readonly="false"
                    data-required="true"
                    placeholder="eg: john@johndoe.com"
                  />
              </lightning-input>
              <lightning-radio-group
                class="immediate-assistance-residence-damaged"
              >
                #shadow-root(open)
                  <span
                    aria-label="Was your primary residence severely damaged by Hurricane Harvey?"
                    data-name="residenceDamaged"
                    data-options="[object Object],[object Object]"
                    data-required="true"
                    data-value="true"
                  />
              </lightning-radio-group>
              <div
                class="slds-form-element__help"
              >
                “Primary residence” is the place you live the majority of the year. “Severely damaged” means your residence had water above your electrical outlets, damage to the roof over main rooms in your home, or the foundation of your home shifted or cracked.
              </div>
              <lightning-radio-group
                class="immediate-assistance-displaced"
              >
                #shadow-root(open)
                  <span
                    aria-label="Were you displaced by the impact of Hurricane Harvey?"
                    data-name="displaced"
                    data-options="[object Object],[object Object]"
                    data-required="true"
                    data-value="true"
                  />
              </lightning-radio-group>
              <div
                class="slds-form-element__help"
              >
                “Displaced” means unable to sleep in your home for more than three nights.
              </div>
              <lightning-radio-group
                class="immediate-assistance-assistance"
              >
                #shadow-root(open)
                  <span
                    aria-label="Do you need emergency financial assistance?"
                    data-name="assistance"
                    data-options="[object Object],[object Object]"
                    data-required="true"
                    data-value="true"
                  />
              </lightning-radio-group>
              <div
                class="slds-form-element__help"
              >
                To replace items such as food, clothes, school supplies or newborn/infant supplies as a result of Hurricane Harvey?
              </div>
              <lightning-button>
                #shadow-root(open)
                  <button>
                    Submit
                  </button>
              </lightning-button>
            </div>
          </div>
          <div
            class="slds-col slds-size_4-of-12"
          />
        </div>
      </div>
    </div>
</community_flashhelp-form>
`;
